support/netutil/netutil.go

```diff
@@ -72,3 +72,64 @@ func watchSignalWithinTimeout(timeout time.Duration, errSignal chan struct{}, ac
 	}
 	action()
 }
+
+// get the local area network IP address
+func GetLocalHost() string {
+	addresses, err := net.InterfaceAddrs()
+	if err != nil {
+		fmt.Printf("net.InterfaceAddrs failed, err: %v", err.Error())
+		return ""
+	}
+
+	for _, address := range addresses {
+		if ipNet, ok := address.(*net.IPNet); ok && !ipNet.IP.IsLoopback() {
+			// IPv4
+			if ipNet.IP.To4() != nil {
+				return ipNet.IP.String()
+			}
+			// IPv6
+			if ipNet.IP.To16() != nil && ipNet.IP.To4() == nil {
+				return ipNet.IP.String()
+			}
+		}
+	}
+	fmt.Printf("no network interface found")
+	return ""
+}
+
+// provide default values for host and port
+func GetHostAndIP(bindStr string, portStr string) (host string, port int) {
+	host = "localhost"
+	localIp := GetLocalHost()
+	validHosts := map[string]bool{
+		"localhost": true,
+		"0.0.0.0":   true,
+		localIp:     true,
+		"127.0.0.1": true,
+	}
+	if _, ok := validHosts[bindStr]; ok {
+		host = bindStr
+	}
+
+	if portStr == "" {
+		port = 7070
+	} else {
+		parsePort, err := strconv.ParseInt(portStr, 10, 64)
+		if err != nil {
+			port = 7070
+		}
+		port = int(parsePort)
+	}
+	return
+}
+
+// construct URL based on host and port
+func GetAndDisplayUrl(host string, port int) string {
+	url := fmt.Sprintf("http://%s:%d", host, port)
+	fmt.Println("Server listen at:")
+	fmt.Printf("-  Local: http://%s:%d \r\n", host, port)
+	if host == "0.0.0.0" {
```

Comment:
I don't think this is valid hereif someone provides a bind to his public IP you won't display it.

You will have to create a function to test if the bind is purely local I think and use icon other method, or copy paste the code here

Reply:
Indeed, it only displays something if not listening on everything